src/layouts/layout.jsx

```diff
@@ -97,11 +97,13 @@ function Layout({ children }) {
             onClick={() => setCategory(filter.name)}
             className={classnames(styles.filterElement, "text__filter__grayFive", filter.name === category && styles.selectedLink)} 
             to={filter.name === 'all' ? '/' : `/categories/${filter.name}/`}>
-            {filter.displayName}
+              {filter.displayName}
           </Link>
         )}
     </div>
   );
+
+  category === undefined && setCategory("all");
```

Comment:
State updates can't be done in the render function. What case the category is `undefined` in?

Reply:
I don't have a concrete answer, but, when I open a blog, and then click on the Xmartlabs logo (goes to the home page) the category is not selected. I did a console.log of category, and, in that scenario, I think, is the only case that category is undefined.

Reply:
Also, if I have many renderers, Gtasby throws an error, in this case, it does not. But I don't know if my "solution" will have any effect on that.

Reply:
Maybe you can check that the case in which the category is not defined is the one in line 76. if it is that, you can add `|| 'all'` at the end of the statement